### ISSUE TITLE
fx: unable to symbolically trace BatchNorm2d due to control flow

### ISSUE THREAD

**Issue by vkuzo** [pytorch/pytorch#44802]
Trying to symbolically trace a vanilla BN leads to errors due to control flow in dim validity checking:

```
import torch

bn = torch.nn.BatchNorm2d(1)
gm = torch.fx.symbolic_trace(bn)
print(gm)

...

    281     def _check_input_dim(self, input):
--> 282         if input.dim() != 4:
    283             raise ValueError('expected 4D input (got {}D input)'
    284                              .format(input.dim()))
...
---> 67         raise TraceError('symbolically traced variables cannot be used as inputs to control flow')

# full trace: https://gist.github.com/vkuzo/5f6141753cb9fba43ea920af0cc0fd02
```

**Comment by vkuzo** [comment]
cc @jamesr66a

**Comment by jamesr66a** [comment]
The expectation for standard library modules in `torch.nn` is that they are traced as leaf modules and that is the default behavior in `Tracer`. What's the background here?

**Comment by vkuzo** [comment]
I see, thanks, makes sense.  The use case we are working with is hitting this because a custom child of `nn.BatchNorm is calling `super().forward(x)`:

```
class BatchNorm2d(torch.nn.BatchNorm2d):
    def forward(self, x):
        if x.numel() > 0:
            return super(BatchNorm2d, self).forward(x)
        # get output shape
        output_shape = x.shape
        return _NewEmptyTensorOp.apply(x, output_shape)
```

since we chatted offline that we can likely remove the `x.numel() > 0` check, this behavior can be deleted as well.  I'll follow up on this specific case.

**Comment by vkuzo** [comment]
this was originally added to user code as a workaround to conv/interpolate/bn ops not handling empty batch correctly.  This has been fixed in PyTorch and the workaround is no longer needed.
